sushichef.py

```diff
@@ -0,0 +1,245 @@
+    topic_contents = {}
+    initial_topics = []
+
+    prev_topic = ""
+
+    counter = 1
+    total_loop = len(lang_data)
+    for i in lang_data:
+        total_loop -= 1
+        try:
+            title = i.text.rstrip().strip()
+            # pp.pprint(title)
+
+            vid_link = ""
+            
+            try:
+                vid_link = i.a.get("href")
+                link_details = {}
+                link_details['video_link'] = vid_link
+                link_details['video_title'] = title
+                initial_topics.append(link_details)
+                # return
+            except:
+                # pp.pprint(i.strip())
+                pass
+
+            if counter == 1:
+                if ":" in title:
+                    counter = 0
+                    prev_topic = title
+
+            if vid_link == "":
+                if ":" in title:
+                    topic_contents[prev_topic] = initial_topics
+                    prev_topic = title
+                    initial_topics = []
+                    pp.pprint(title)
+            
+            pass
+        except:
+            # pp.pprint(i.strip())
+            pass
+
+        if total_loop == 0:
+            topic_contents[prev_topic] = initial_topics
+    return topic_contents
+
+def scrape_page(url):
+    pp = pprint.PrettyPrinter()
+
+    # TODO: check if local html file exist if not create it
+    # Online scrape
+    response = requests.get(url)
+    page = BeautifulSoup(response.text, 'html5lib')
+
+    # if not os.path.exists(TEMP_HTML_FILE):
+    #     file = open(TEMP_HTML_FILE, "w") 
+    #     file.write(page.prettify())
+    #     file.close() 
+
+    # Offline scrape
+    # response = urllib.request.urlopen(LOCAL_HTML_PATH)
+    # page = BeautifulSoup(response, 'html5lib')
+
+    content_divs = page.body.div
+
+    list_divs = list(content_divs.children)
+    language_divs = list(list_divs[5].children)
+    # language divs is a list of language just
+
+    # To get the first language append four to the language counter
+    language_counter = 7
+    get_language = list(language_divs[language_counter])
+    # create_language_data(get_language)
+    # pp.pprint(list(language_divs)[7])
+    lang_name = language_divs[language_counter].get('id')
+    data_contents = { lang_name: create_language_data(get_language) }
+
+    # pp.pprint(data_contents)
+        
+
+def scrape_video_data(url ,title):
+
+    params = {"format": "json", "url": url}
+    url_embed = "https://www.youtube.com/oembed"
+    query_string = urllib.parse.urlencode(params)
+    url = url_embed + "?" + query_string
+    video_data = {}
+    print(url)
+    with urllib.request.urlopen(url) as response:
+        response_text = response.read()
+        video_data = json.loads(response_text.decode())
+
+    video_data = []
+    pp = pprint.PrettyPrinter()
+    try:
+        video = ArvindVideo(
+                    url=src, 
+                    title=title, 
```

Comment:
The titles of the videos from YouTube all seem to have the following repeating structure:
`{{video_name}} - {{language}} - {{filesize}.avi`, e.g. `BRAIN CAP - ENGLISH - 19MB.avi`.

It would be preferable to get rid of the `- {{language}} - {{filesize}.avi` and leave just the video name.

Reply:
I already implemented it, Thanks.